jina/executors/encoders/multimodal/__init__.py

```diff
@@ -16,22 +15,30 @@ class BaseMultiModalEncoder(BaseExecutor):
     """
 
     def __init__(self,
-                 position_by_modality: Dict[str, int] = {},
+                 position_by_modality: List[str] = [],
                  *args,
                  **kwargs):
         """
-        :param position_by_modality: the map of arguments indicating in which order the modalities they need to come
+        :param position_by_modality: the list of arguments indicating in which order the modalities they need to come
         for the encoding method
         :return:
         """
         super().__init__(*args, **kwargs)
-        self.position_by_modality = position_by_modality
+        self._position_by_modality = position_by_modality
+
 
-    # @batching
-    @as_ndarray
     def encode(self, *data: 'np.ndarray', **kwargs) -> 'np.ndarray':
         """
         :param: data: M arguments of shape `B x (D)` numpy ``ndarray``, `B` is the size of the batch, `M` is the number of modalities
         :return: a `B x D` numpy ``ndarray``
         """
         raise NotImplementedError
+
+    @property
```

Comment:
Remove it from executor and move thr attribute to driver

Reply:
ah I misunderstood.. I'll move it back